### PR TITLE
Add Ncycle Iterator

### PR DESCRIPTION
**Pull request by lstagner**

### CODE REVIEW

**Comment by iamed2** [comment]
Please add tests for:
- `collect(ncycle(...))` (this will test various iterator machinery)
- `eltype(ncycle(...))`

File: src/Iterators.jl [hunk 1]
```diff
@@ -612,6 +613,35 @@ next{T}(r::PeekIter{UnitRange{T}}, i) = next(r.it, i)
...
+"""
+    ncycle(iter, n)
+An iterator that cycles through `iter` `n` times.
+"""
+ncycle(iter, n::Int) = NCycle(iter, n)
+
+eltype(nc::NCycle) = eltype(nc.iter)
+length(nc::NCycle) = nc.n*length(nc.iter)
+size(nc::NCycle) = (size(nc.iter)..., nc.n)
+iteratorsize{I}(::Type{NCycle{I}}) = iteratorsize(I)
```
Comment (iamed2) [inline src/Iterators.jl]: (that would be a good test to add if `flatten` was exported from Base...)

**Comment by lstagner** [comment]
Sorry for the delay. I believe I have address your concerns.

**Comment by iamed2** [comment]
Please fix the test failures. In particular, don't use flatten; `flatten(x)` can be changed to `chain(x...)`.

**Comment by iamed2** [comment]
Travis failure is unrelated now!

**Pull request merged**